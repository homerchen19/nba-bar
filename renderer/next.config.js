const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
