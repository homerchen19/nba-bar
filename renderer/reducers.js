import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
