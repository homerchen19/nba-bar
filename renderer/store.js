import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
