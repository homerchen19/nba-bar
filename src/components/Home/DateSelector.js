import React from 'react';
import PropTypes from 'prop-types';
import { Flex, Icon } from 'antd-mobile';
import styled from 'styled-components';
import format from 'date-fns/format';
import startOfDay from 'date-fns/start_of_day';

import { colors } from '../../styles/theme';

const formatDate = date => format(startOfDay(date), 'DD, MMM, YYYY');

const Wrapper = styled.div`
  position: fixed;
  z-index: 999;
  width: 100%;
  height: 30px;
  padding: 4px 7px;
  background-color: ${colors.darkBlue};
  color: ${colors.white};
`;

const Button = styled(Icon)`
  cursor: pointer;
`;

const Item = styled(Flex.Item)`
  margin-left: 0 !important;
  text-align: center;
`;

const Date = styled.p`
  font-size: 15px;
`;

const DateSelector = ({ date, addDay, subDay }) => (
  <Wrapper>
    <Flex>
      <Button type="left" onClick={subDay} />
      <Item>
        <Date>{formatDate(date)}</Date>
      </Item>
      <Button type="right" onClick={addDay} />
    </Flex>
  </Wrapper>
);

DateSelector.propTypes = {
  addDay: PropTypes.func.isRequired,
  date: PropTypes.number.isRequired,
  subDay: PropTypes.func.isRequired,
};

export default DateSelector;
