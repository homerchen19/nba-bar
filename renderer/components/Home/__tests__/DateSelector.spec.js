import React from 'react';
import renderer from 'react-test-renderer';

import DateSelector from '../DateSelector';

it('be defined', () => {
  expect(DateSelector).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    date: 1522984495289,
    addDay: jest.fn(),
    subDay: jest.fn(),
  };
  const tree = renderer.create(<DateSelector {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
