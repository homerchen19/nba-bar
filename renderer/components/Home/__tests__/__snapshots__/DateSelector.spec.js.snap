// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="DateSelector__Wrapper-sc-1doghr4-0 NtdVW"
>
  <div
    className="am-flexbox am-flexbox-align-center"
  >
    <svg
      className="DateSelector__Button-sc-1doghr4-1 gofxkt am-icon am-icon-left am-icon-md"
      onClick={[MockFunction]}
    >
      <use
        xlinkHref="#left"
      />
    </svg>
    <div
      className="am-flexbox-item DateSelector__Item-sc-1doghr4-2 QPjgb"
    >
      <p
        className="DateSelector__Date-sc-1doghr4-3 gRtgOB"
      >
        06, Apr, 2018
      </p>
    </div>
    <svg
      className="DateSelector__Button-sc-1doghr4-1 gofxkt am-icon am-icon-right am-icon-md"
      onClick={[MockFunction]}
    >
      <use
        xlinkHref="#right"
      />
    </svg>
  </div>
</div>
`;
